### PR TITLE
rgw: improve support for Swift's URL schema with account name inside.

### PR DESCRIPTION
**Pull request by rzarzynski**

### CODE REVIEW

File: src/rgw/rgw_common.h [hunk 1]
```diff
@@ -1048,6 +1048,7 @@ struct req_state {
    const char *os_auth_token;
    string swift_user;
    string swift_groups;
+   string swift_account;
```
Comment (rzarzynski) [inline src/rgw/rgw_common.h]: I'm not sure whether the scope should be extended right now. However, the access to the tenant name will be very useful for auth mechanisms when it is impossible to get the tenant basing only on _X-Token-Auth_.

**Comment by yehudasa** [comment]
@rzarzynski I have a pending branch that should handle the account stuff differently (wip-5073-3), and it's part of our plan to complete this soon(is)

**Comment by rzarzynski** [comment]
@yehudasa goal of this patch was to allow using URL format with account/tenant name inside (even if the part would be simply stripped for now). _rgw_swift_tenant_name_ works a bit different. It enforces presence of a single, specified name (inc. the _AUTH__ prefix) within requested URL: https://github.com/rzarzynski/ceph/blob/wip-11163/src/rgw/rgw_rest_swift.cc#L942.

It would be great to see multi-tenancy in RGW. To be honest, this PR was a first step from my side to start working on them. :-) I will definitely take a look on wip-5073-3. Thanks!

**Comment by zaitcev** [comment]
This looks at best incomplete to me.

First, it's obvious that req_state.swift_account is not used at this point. Radoslaw himself wrote a note "not sure whether the scope should be extended right now". However, the merit of the thing cannot be determined until the use appears. I think the patch should include such use.

Second, the same thing applies to rgw_swift_tenant_in_url itself too. If you set it today, the auth code must start returning the StorageURL set where the account is present. Currently, I do not see it doing that.

**Pull request closed without merging**